flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServer.java

```diff
@@ -70,18 +82,65 @@
 	private final int maxConnections;
 
 	/**
+	 * Shutdown hook thread to ensure deletion of the storage directory (or <code>null</code> if
+	 * {@link RecoveryMode#STANDALONE})
+	 */
+	private final Thread shutdownHook;
+
+	/**
 	 * Instantiates a new BLOB server and binds it to a free network port.
 	 * 
 	 * @throws IOException
 	 *         thrown if the BLOB server cannot bind to a free network port
 	 */
 	public BlobServer(Configuration config) throws IOException {
+		checkNotNull(config, "Configuration");
+
+		this.recoveryMode = RecoveryMode.fromConfig(config);
 
 		// configure and create the storage directory
 		String storageDirectory = config.getString(ConfigConstants.BLOB_STORAGE_DIRECTORY_KEY, null);
 		this.storageDir = BlobUtils.initStorageDirectory(storageDirectory);
 		LOG.info("Created BLOB server storage directory {}", storageDir);
 
+		if (recoveryMode == RecoveryMode.STANDALONE) {
+			recoveryBasePath = null;
+		}
+		else {
+			// Initialize file state backend for recovery
+			String stateBackend = config.getString(ConfigConstants.STATE_BACKEND,
+					ConfigConstants.DEFAULT_STATE_BACKEND);
+
+			if (!stateBackend.toLowerCase().equals("filesystem")) {
+				throw new IllegalConfigurationException(String.format("Illegal state backend " +
+								"configuration '%s'. Please configure 'FILESYSTEM' as state " +
+								"backend and specify the recovery path via '%s' key.",
+						stateBackend, ConfigConstants.STATE_BACKEND_FS_RECOVERY_PATH));
+			}
```

Comment:
State backend enum is part of the streaming module, which is not a dependency of the runtime atm. I will coordinate with @aljoscha and check what a good place for it is with the streaming rewrite going on.